sdcflows/interfaces/bspline.py

```diff
@@ -124,27 +130,48 @@ class BSplineApprox(SimpleInterface):
 
     def _run_interface(self, runtime):
         from sklearn import linear_model as lm
-        from scipy.sparse import vstack as sparse_vstack
+
+        # Output name baseline
+        out_name = fname_presuffix(
+            self.inputs.in_data, suffix="_field", newpath=runtime.cwd
+        )
 
         # Load in the fieldmap
         fmapnii = nb.load(self.inputs.in_data)
-        data = fmapnii.get_fdata(dtype="float32")
+        zooms = fmapnii.header.get_zooms()
 
-        # Generate the output naming base
-        out_name = fname_presuffix(
-            self.inputs.in_data, suffix="_field", newpath=runtime.cwd
+        # Get a mask (or define on the spot to cover the full extent)
+        masknii = (
+            nb.load(self.inputs.in_mask)
+            if isdefined(self.inputs.in_mask)
+            else None
         )
-        # Create a copy of the header for use below
-        hdr = fmapnii.header.copy()
-        hdr.set_data_dtype("float32")
 
+        need_resize = np.any(np.array(zooms) < self.inputs.zooms_min)
+        if need_resize:
+            from sdcflows.utils.tools import resample_to_zooms
+
+            zooms_min = np.maximum(zooms, self.inputs.zooms_min)
```

Comment:
Just for clarity, I think we want to rename from `zooms_min`, since it now diverges form `self.inputs.zooms_min`.

Reply:
what would you suggest? (I have no particular preference)

Reply:
`target_zooms` made sense to me.